src/modules/app/VirtualMouseModule.js

```diff
@@ -21,11 +21,14 @@ export class VirtualMouseModule extends Events {
     this.patchEvents = patchEvents;
   }
 
-  update(e) {
+  update(e, customX, customY) {
     const rect = this.canvas.getBoundingClientRect();
 
-    this.mouse.x = ((e.clientX - rect.left) / (rect.right - rect.left)) * 2 - 1;
-    this.mouse.y = -((e.clientY - rect.top) / (rect.bottom - rect.top)) * 2 + 1;
+    const x = customX || e.clientX;
+    const y = customY || e.clientY;
+
+    this.mouse.x = ((x - rect.left) / (rect.right - rect.left)) * 2 - 1;
```

Comment:
If `rect = { left: 0, right: 100 }` and `mouse.x = 50`, then we have

`((50-0) / (100-0)) * 2 -1 = (50 / 100) * 2 - 1 = 0.5 * 2 - 1 = 1 - 1 = 0`

so we get mouse coords normalized in `[-1, 1]` space